File: TrainApp/src/service/tiqueteService.js
import axios from 'axios';

const API_URL = 'http://localhost:5173/api/tiquete';

export const comprarTiquete = async (tiquete) => {
    try {
        const response = await axios.post(API_URL, tiquete);
        return response.data;
    } catch (error) {
        console.error('Error buying tiquete:', error);
        throw error;
    }
};

export const getTiquetes = async () => {
    try {
        const response = await axios.get(API_URL);
        return response.data;
    } catch (error) {
        console.error('Error fetching tiquetes:', error);
        throw error;
    }
};

File: Server/TrainApi/Service/TiqueteService.cs
using System.Collections.Generic;

public class TiqueteService : ITiqueteService
{
    private readonly List<Tiquete> _tiquetes = new();

    public void ComprarTiquete(Tiquete tiquete)
    {
        _tiquetes.Add(tiquete);
    }

    public IEnumerable<Tiquete> GetTiquetes()
    {
        return _tiquetes;
    }
}
